### DISCUSSION TITLE
How to parse the eventKeyJmesPath field and validate a key on the inside?

### DISCUSSION THREAD

**Discussion by gabrielleandro0801** [aws-powertools/powertools-lambda-typescript#2029] [General]
Hey there!

I read that the Java Powertools [lib](https://docs.powertools.aws.dev/lambda/python/latest/utilities/jmespath_functions/) contains the powertools_json function to parse a string to JSON, so I can check the idempotency through any key inside the JSON (Image 1). And so does the Python Powertools [lib](https://docs.powertools.aws.dev/lambda/java/utilities/serialization/#jmespath-functions) (Image 2).
Is there any equivalent for these functions in Typescript Powertools lib or should I parse the JSON another way?
 
Thanks for your attention!

![image](https://github.com/aws-powertools/powertools-lambda-typescript/assets/71289016/dcc15dfd-ca12-4b8d-a03e-1437c4846aa9)
![image](https://github.com/aws-powertools/powertools-lambda-typescript/assets/71289016/47d3ac4a-0587-41bf-b2a2-4acb0e78f94c)

**Comment by dreamorosi** [comment]
Hi @gabrielleandro0801 thank you for opening this discussion and for the interest in Powertools!

At the moment we don't have an equivalent for this feature but we are actively working on this item and should be able to share more details on when it'll be published soon.

The feature is tracked in #1298 and it's dependant on us completing the implementation for our own JMESPath module.
For added context: the official `jmespath` module [published on npm](https://www.npmjs.com/package/jmespath) does not support registering custom functions, this prevents us from offering custom functions like JSON and base64 deserialization which [are not part of the JMESPath specification](https://jmespath.org/specification.html#built-in-functions).

With this in mind, and considering that the official module was implemented following older practices as well as a complete lack of TypeScript support, we have decided to implement our own JMESPath utility. The work on this item is [in advanced state](https://github.com/aws-powertools/powertools-lambda-typescript/tree/feat/jmespath) and we are currently focused on reaching 100% compliance with the JMESPath specification (aka 100% of test suite passing).

In the meanwhile, it would be immensely helpful for us if you could please leave a 👍 under the issue for this feature (#1298) and maybe leaving a comment expressing your interest (or simply linking this discussion). We constantly adjust our roadmap and prioritization largely happens as a result of customer demand.

**Comment by gabrielleandro0801** [comment]
Hi, @dreamorosi!

Thanks for your answer!
Now I could understand the situation. I already left my :+1: and a comment in https://github.com/aws-powertools/powertools-lambda-typescript/issues/1298.
No problems, I will be waiting for the feature to be released.

Best regards!

**Reply by dreamorosi** [reply]
Hi @gabrielleandro0801, we have just launched this feature in the latest release [v2.1.0](https://github.com/aws-powertools/powertools-lambda-typescript/releases/tag/v2.1.0) - let us know if you try it and find any issues!

**Reply by gabrielleandro0801** [reply]
Hi, @dreamorosi
Thank you very much for the feedback! :smile: 

Sure, I'll try it and I'll tell you if anything goes wrong.

**Reply by gabrielleandro0801** [reply]
Hi, @dreamorosi 
I've just tried the new feature and it worked perfectly.
Thank you, once again!
